tests/rbd/delete_clones_with_io.py

```diff
@@ -0,0 +1,105 @@
+import time
+
+from krbd_io_handler import krbd_io_handler
+from rbd.exceptions import RbdBaseException
+
+from ceph.parallel import parallel
+from tests.rbd.rbd_utils import Rbd
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(**kw):
+    """Delete Clones while IO is running on parent.
+
+    Args:
+        kw: test data
+
+    Returns:
+        int: The return value. 0 for success, 1 otherwise
+
+    Pre-requisites :
+    We need atleast one client node with ceph-common package,
+    conf and keyring files
+
+    Test cases covered -
+    1) CEPH-9225 - RBD-Kernel_Client: Delete all Clone at once
+
+    Test Case Flow
+    1. Create a pool, an Image, 10 snapshots, 10 clones.
+    2. generate IO on the Image, in parallel delete the clones.
+
+    """
+    log.info("Running test - delete clones with IOs")
+
+    scenarios = [Rbd(**kw)]
+
+    kw["config"]["ec-pool-k-m"] = "go_with_default"
+    scenarios.append(Rbd(**kw))
+
+    config = kw["config"]
+
+    pools = ["test_rbd_ec_pool_scenario", "test_rbd_rep_pool"]
+    image = config.get("image_name", "rbd_test_image")
+    size = "10G"
+
+    for rbd in scenarios:
+        pool = pools.pop()
+        kw["rbd_obj"] = rbd
+        try:
+
+            def create_clone(*args):
+                if not rbd.create_pool(poolname=pool):
```

Comment:
Why not use the initial_rbd_config method for pool and image creation?

Reply:
Currently initial_rbd_config doesn't support `no config - all default arguments mode`
I've draft changes for them, will raise a PR in 2-3 days.

Can we please merge this PR (to keep tier-1 100% goal intact for Q4-22) ?

Reply:
Sure that should be okay.